bookwyrm/management/commands/generate_preview_images.py

```diff
@@ -0,0 +1,49 @@
+""" Generate preview images """
+import sys
+
+from django.core.management.base import BaseCommand
+
+from bookwyrm import activitystreams, models, settings, preview_images
+
+
+def generate_preview_images():
+    """generate preview images"""
+    print("   | Hello! I will be generating preview images for your instance.")
+    print(
+        "🧑‍🎨 ⎨ This might take quite long if your instance has a lot of books and users."
+    )
+    print("   | ✧ Thank you for your patience ✧")
+
+    # Site
+    sys.stdout.write("   → Site preview image: ")
+    preview_images.generate_site_preview_image_task()
+    sys.stdout.write(" OK 🖼\n")
+
+    # Users
+    users = models.User.objects.filter(
+        local=True,
+        is_active=True,
+    )
+    sys.stdout.write("   → User preview images ({}): ".format(len(users)))
+    for user in users:
+        preview_images.generate_user_preview_image_task(user.id)
+        sys.stdout.write(".")
+    sys.stdout.write(" OK 🖼\n")
+
+    # Books
+    books = models.Book.objects.select_subclasses().filter()
+    sys.stdout.write("   → Book preview images ({}): ".format(len(books)))
+    for book in books:
+        preview_images.generate_edition_preview_image_task(book.id)
+        sys.stdout.write(".")
+    sys.stdout.write(" OK 🖼\n")
```

Comment:
1) I love your use of emoji here, this is a work of art. 2) what's the motivation for switching between `print` and `sys.stdout`? Not a criticism, I just genuinely don't know

Reply:
1. Thanks! I figured that, as long as we're seeing this screen for a long time (this command doesn't send the task to Celery, but perhaps it should? perhaps as an argument?), it should be nice to see
2. `print()` inserts a `\n` at the end while `sys.stdout` does not, so I used `print()` for full lines, and `sys.stdout` to add one (or a few) characters at the end of the current line

Reply:
I've now switched to `BaseCommand`'s `self.stdout` (as the documentation suggested). Now, each `write()` call must specify `ending=""` if I don't want a `\n` added at the end of the string.